File: src/server/game/config/win_conditions.go
package config

type WinCondition struct {
	Name         string        `json:"name"`
	Requirements []Requirement `json:"requirements"`
}

type WinConditionsConfig struct {
	WinConditions []WinCondition `json:"win_conditions"`
}

var (
	winConditionsConfig sync.Once
	WinConditionsConfigVar WinConditionsConfig
)

// GetWinConditionsConfig allows to get the game configuration linked to win conditions globally.
func GetWinConditionsConfig() WinConditionsConfig {
	return WinConditionsConfigVar
}

File: src/game/rules/win_conditions.lua
local WinCondition = {}
WinCondition.__index = WinCondition

function WinCondition:new(name, requirements)
    local winCondition = setmetatable({}, WinCondition)

    winCondition.name = name
    winCondition.requirements = requirements or {}

    return winCondition
end

return WinCondition
